hpctestlib/system/fs/trivy_scan.py

```diff
@@ -0,0 +1,271 @@
+
+    @run_after('init')
+    def download_trivy_if_required(self):
+        if not self.download_trivy:
+            return
+
+        # using GitHub API in case the version is set to latest
+        # it may hit the maximum number of API calss to GitHub
+        if self.trivy_release == 'latest':
+            self.prerun_cmds = [
+                f"__tar_file=`curl -s https://api.github.com/repos/aquasecurity/trivy/releases/latest | grep browser_download_url | grep _{self.trivy_os_arch}.tar.gz\\\" | awk '{{print $2}}' | tr -d '\"'`",  # noqa: E501
+                r'curl -LJO ${__tar_file}',
+                r"tar xf $(basename ${__tar_file})",
+                r"rm -f $(basename ${__tar_file})"
+            ]
+        else:
+            # downloading from the release url if version is not set to latest
+            tar_file = f'trivy_{self.trivy_release}_{self.trivy_os_arch}.tar.gz'
+            self.prerun_cmds = [
+                f'curl -LJO https://github.com/aquasecurity/trivy/releases/download/v{self.trivy_release}/{tar_file}',  # noqa: E501
+                f"tar xf {tar_file}",
+                f"rm -f {tar_file}"
+            ]
+
+        self.executable = f'./{self.executable}'
+        self.postrun_cmds = [
+            f'rm -rf {self.executable}'
+        ]
+
+    @run_after('init')
+    def prepend_fs_option_cmd_opts(self):
+        self.executable_opts = ['fs', f'{self.folders}'] + self.executable_opts
+
+    @run_before('run')
+    def update_executable_opts(self):
+        self.executable_opts += [
+            '--timeout', self.time_limit if self.time_limit else '5000m',
+            '--cache-dir', self.stagedir,
+            '-f', 'json',
+            '-o', self.trivy_output_file
+        ]
+
+    @run_before('run')
+    def set_scanner(self):
+        self.executable_opts += [
+            '--scanners', f'{self.__scanner_type}'
+        ]
+
+    def load_json_output(self, filename):
+        with open(filename) as fp:
+            try:
+                trivy_output_json = json.loads(fp.read())
+                return trivy_output_json
+            except json.JSONDecodeError as e:
+                return None
+
+    @deferrable
+    def assert_supported_schemaversion(self, trivy_output_json):
+        field_name = 'SchemaVersion'
+        return sn.and_(
+            sn.assert_in(field_name, trivy_output_json,
+                         msg=f'{field_name} not found in trivy output. You may '
+                             f'have a corrupt {self.trivy_output_file} file'),
+            sn.assert_eq(self.trivy_json_schema_version,
+                         trivy_output_json[field_name],
+                         msg=f'{field_name} value not supported. Found '
+                             f'{trivy_output_json[field_name]}, expected '
+                             f'{self.trivy_json_schema_version}')
+        )
+
+    @deferrable
+    def config_sanity_check(self, trivy_output_json):
```

Comment:
I would name all these function as `assert_XXX` and construct the name below accordingly.